scripts/types/database/public/Dossier.js

```diff
@@ -5,6 +5,20 @@ export {};
  * Identifier type for public.dossier
  * @typedef {number & { __brand: 'DossierId' }} DossierId
  */
+
+/** Possible values for public.dossier.phase
+ * @typedef {"accompagnement amont" | "accompagnement amont terminé" | "instruction" | "décision" | "refus tacite" | null} DossierPhase
```

Comment:
Ici et comme pour le fichier `.ts` j'ai voulu fixer les valeurs possibles pour ces trois colonnes. Lorsque je lance `npm run build-db-types`, ça me passe tout en simple string.

Je veux bien ton avis là-dessus, qu'est-ce qui est le mieux de faire ?

Reply:
Oui, c'est un problème qu'on a qui est que l'outil qui récupère les types de la base de données nous dit "string" alors que nous on sait que c'est plus restreint que ça 

Ptèt on peut créer un type enum ?
https://knexjs.org/guide/schema-builder.html#enum-enu

je ne sais pas comment `npm run build-db-types` va réagir

Il devrait être possible aussi pour nous de modifier/surcharger les types générés automatiquement (sans modifier les fichiers eux-même), mais je n'ai pas trop creusé comment faire encore

Reply:
Ptèt avec un truc genre https://medium.com/@ofirstiberdev/overriding-type-properties-in-typescript-like-a-pro-54203a817253

Reply:
> Ptèt on peut créer un type enum ?
> https://knexjs.org/guide/schema-builder.html#enum-enu

Comme ça, en lisant la migration, j'ai l'impression que c'est un peu rigide de gérer l'enum dans PostGres si on doit faire une migration à chaque fois qu'on veut ajouter/changer une valeur. Je vais regarder pour une solution pour surcharger nos types.